src/main/java/org/testng/ITestRunnerFactory2.java

```diff
@@ -0,0 +1,28 @@
+package org.testng;
+
+import org.testng.xml.XmlTest;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A factory for TestRunners to be used by SuiteRunners.
+ */
+public interface ITestRunnerFactory2 extends ITestRunnerFactory {
```

Comment:
I am not sure about it. I was of the opinion that eventually when TestNG moves to JDK8, then `ITestRunnerFactory2` can be gotten rid of and its implementation could very well reside in `ITestRunnerFactory` as a `default` method.